src/site/content/en/blog/serial/index.md

```diff
@@ -253,6 +253,21 @@ while (true) {
 }
 ```
 
+To specify the buffer into which data is read instead of the stream allocating a buffer for each chunk, get a [ReadableStreamBYOBReader] by calling `port.readable.getReader({ mode: "byob" })` and read data into a new `ArrayBuffer`. Note that the Web Serial API supports "Bring Your Own Buffer" readers in Chrome 106 or later.
+
+```js
+try {
+  const reader = port.readable.getReader({ mode: "byob" });
+  let buffer = new ArrayBuffer(512);
+  // Call reader.read() to read data into the buffer...
```

Comment:
Can you include some examples of how to call `read()` with a BYOB reader? When writing the tests I encountered a few gotchas:

1. You have to pass an `ArrayBufferView` (like `Uint8Array`) rather than a raw ArrayBuffer.
2. The buffer is transferred so if you want to reuse it (which is half the point) you need to extract it out of `value.buffer`.

See the `readInto()` function from https://github.com/web-platform-tests/wpt/blob/master/serial/serialPort_readable_byob.https.any.js#L5 for an example of how to use a BYOB reader to read a specific amount of data from the port.

See https://github.com/GoogleChromeLabs/serial-terminal/blob/main/src/index.ts#L271 for how I implemented fallback and reuse of the buffer.

Reply:
Thanks for sharing those examples. I've added some.